web/packages/core/src/ruffle-player.ts

```diff
@@ -667,6 +669,38 @@ export class RufflePlayer extends HTMLElement {
         }
     }
 
+    /**
+     * Async helper function for fetching the SWF.
+     */
+    private async fetchSWF(): Promise<void> {
```

Comment:
There's no need for a separate `fetchSWF` helper function, `downloadSWF` can be `async` itself and have the same logic in it.